### DISCUSSION TITLE
Cant integrate with home assistant

### DISCUSSION THREAD

**Discussion by bugac** [raomin/ESPAltherma#302] [Q&A] [answered]
So far im spending my 3 day, on this noob problem, cant find answer online. because I feel like I'm missing something, and I'm the only one in the world facing this problem
i made to work my M5StickC with my daikin altherma 3

![image](https://github.com/raomin/ESPAltherma/assets/404278/8d634793-72da-4966-85d1-5c2aa15a8ea5)

but it is imposible to add these atributes as sensors to HA.

Found git example from here:

`sensor:
  - platform: template
    sensors:
      espaltherma_outdoorairtemp:
        friendly_name: "Outdoor air temp"
        value_template: "{{ state_attr('sensor.althermasensors','Außenlufttemp.') }}"
        unit_of_measurement: '°C'
      espaltherma_fan:
        friendly_name: "Fan 1"
        value_template: "{{ state_attr('sensor.althermasensors','Ventilator 1 (10 rpm)') * 10 }}"
        unit_of_measurement: 'rpm'
      espaltherma_flow:
        friendly_name: "Flow sensor"
        value_template: "{{ state_attr('sensor.althermasensors','Durchflussmenge (l/min)') }}"
        unit_of_measurement: 'l/min'
      espaltherma_refigtemp:
        friendly_name: "Refig temp"
        value_template: "{{ state_attr('sensor.althermasensors','R3T-Kältemittel Flüssigkeitstemperatur') }}"
        unit_of_measurement: '°C'
      espaltherma_inverter:
        friendly_name: "INV current"
        value_template: "{{ state_attr('sensor.althermasensors','INV Primärstrom (A)') }}"
        unit_of_measurement: 'A'
      espaltherma_operationstate:
        friendly_name: "Error type"
        value_template: "{{ state_attr('sensor.althermasensors','Betrieb / Störung') }}"
      espaltherma_voltage:
        friendly_name: "Voltage (V)"
        value_template: "{{ state_attr('sensor.althermasensors','Spannung (V)') }}"
      espaltherma_wateroutlet:
        friendly_name: "Leaving water temp"
        value_template: "{{ state_attr('sensor.althermasensors','Auslass Wasser BUH Temp. (R2T)') }}"
        unit_of_measurement: '°C'
      espaltherma_operation:
        friendly_name: "Operation mode"
        value_template: "{{ state_attr('sensor.althermasensors','Aktuelle Betriebsart') }}"
      espaltherma_iuoperation:
        friendly_name: "Indoor Operation mode"
        value_template: "{{ state_attr('sensor.althermasensors','I/U betriebsart') }}"
      espaltherma_dhw:
        friendly_name: "DHW temp"
        value_template: "{{ state_attr('sensor.althermasensors','R5T-Brauchwassertemperatur im Speicher') }}"
        unit_of_measurement: '°C'
      espaltherma_cop:
        friendly_name: "COP"
        unit_of_measurement: 'COP'
        value_template: "{% if
  state_attr('sensor.althermasensors','Aktuelle Betriebsart') in ('Heating', 'Cooling')
  and state_attr('sensor.althermasensors','Frostschutz') == 'OFF'
  and state_attr('sensor.althermasensors','INV Primärstrom (A)')  | float != 0
%} 
{{ 
  (
    (
      state_attr('sensor.althermasensors','Durchflussmenge (l/min)') | float
      * 0.06
      * 1.16
      *
        (
          state_attr('sensor.althermasensors','Auslass Wasser Wärmetauscher Temp. (R1T)') | float
          - state_attr('sensor.althermasensors','R4T-Wasser Rücklauftemp. vor dem Plattenwärmetauscher') | float
        )
    )
    /
    (
      state_attr('sensor.althermasensors','INV Primärstrom (A)') | float
      * state_attr('sensor.althermasensors','Spannung (V)') | float
      / 1000
    )
  )
  | round(2)
  | abs
}}
{% else %}
0
{% endif %}"`

Added it to configuration.yaml

![image](https://github.com/raomin/ESPAltherma/assets/404278/e3a3fc7b-46f0-4320-82fe-b321114c110e)

and all i got is this - 
![image](https://github.com/raomin/ESPAltherma/assets/404278/2955f182-e602-4caf-8a7a-589264b82a9a)

when im trying to configure any of these i see this message
![image](https://github.com/raomin/ESPAltherma/assets/404278/09fc435f-58e6-469b-9677-c03a2cc80f30)

maybe it's related to my problem that the sensors don't work?

what is wrong?

**Comment by bugac** [comment]
still no answer. Am I the only one having this problem?
Linking esp readings to home assistant UI?

**Comment by raomin** [comment]
Looks like you're linking german attribute names:
![image](https://github.com/raomin/ESPAltherma/assets/4460208/c6d851b4-35f5-475b-9777-5a297ec27622)
With English attribute definition:
![image](https://github.com/raomin/ESPAltherma/assets/4460208/9424be93-3812-437d-b6d5-e404a77ff8cf)

**Reply by raomin** [reply, marked as answer]
your `value_template` should be more like:
`value_template: "{{ state_attr('sensor.altherma','Operation Mode') }}"`
